### ISSUE TITLE
It seems you got one learning rate per epoch.

### ISSUE THREAD

**Issue by ArchNew** [ildoonet/pytorch-gradual-warmup-lr#6]
If I read your code correctly, I think your learning rate computes based on number of epochs, not loops. According to the paper, it seems the learning rate should be computed based on number of loops.

**Comment by ildoonet** [comment]
@ArchNew The parameter ```epoch```can be INT as well as FLOAT.

So you can provide the value like 0.01, 0.02, ... 

In other words, you can calculate the fractional value of epoch which is propotional to steps, and then you can use the value.

Something like..

```python

for epoch in ...
    for step in ...
       curr_epoch = epoch + float(step) / num_steps

```

**Comment by ArchNew** [comment]
Thanks!
